### PR TITLE
Refactor/blender transition window

### PR DESCRIPTION
**Pull request by martiniil**
* minor changes in computing trajectory alignment
* comment fixes

### CODE REVIEW

File: pilz_trajectory_generation/src/trajectory_blender_transition_window.cpp [hunk 1]
```diff
@@ -266,11 +267,10 @@ void pilz::TrajectoryBlenderTransitionWindow::determineTrajectoryAlignment(const
                                                                 std::size_t second_interse_index,
                                                                 std::size_t &blend_align_index) const
 {
-  double tau_1 = (req.first_trajectory->getWayPointDurationFromStart(req.first_trajectory->getWayPointCount()) -
-                  req.first_trajectory->getWayPointDurationFromStart(first_interse_index));
-  double tau_2 = req.second_trajectory->getWayPointDurationFromStart(second_interse_index);
+  size_t way_point_count_1 = req.first_trajectory->getWayPointCount() - first_interse_index;
+  size_t way_point_count_2 = second_interse_index+1;
 
-  if(tau_1 > tau_2)
```
Comment (agutenkunst) [inline pilz_trajectory_generation/src/trajectory_blender_transition_window.cpp:273]: Does this now only work if both trajectories are equally discretized?

**Review by agutenkunst — CHANGES_REQUESTED** [review]
See comments

File: pilz_trajectory_generation/src/trajectory_blender_transition_window.cpp (hunk 1 above)
Comment (martiniil) [reply pilz_trajectory_generation/src/trajectory_blender_transition_window.cpp:273]: Yes, but this was always the case for the blend method. We just did not exploit it here.

Comment (martiniil) [reply pilz_trajectory_generation/src/trajectory_blender_transition_window.cpp:273]: See validateRequest():
https://github.com/PilzDE/pilz_industrial_motion/blob/18a26a448226a68abb51f2ad74352a456d1a5ccf/pilz_trajectory_generation/src/trajectory_blender_transition_window.cpp#L149

Comment (agutenkunst) [reply pilz_trajectory_generation/src/trajectory_blender_transition_window.cpp:273]: ok

**Review by agutenkunst — APPROVED** [review]

**Review by agutenkunst — APPROVED** [review]

File: pilz_trajectory_generation/include/pilz_trajectory_generation/trajectory_blender_transition_window.h [hunk 2]
```diff
@@ -87,24 +88,24 @@ class TrajectoryBlenderTransitionWindow : public TrajectoryBlender
 
   /**
    * @brief Determine how the second trajectory should be aligned with the first trajectory for blend.
-   * tau_1 is the time of the first trajectory from the first_interse_index to the end
-   * tau_2 is the time of the second trajectory from begin to  the second_interse_index
+   * Let tau_1 be the time of the first trajectory from the first_interse_index to the end and tau_2 the time of the
```
Comment (agutenkunst) [inline pilz_trajectory_generation/include/pilz_trajectory_generation/trajectory_blender_transition_window.h:93]: Please check the resulting generated html documentation...

**Review by agutenkunst — CHANGES_REQUESTED** [review]
Please recheck the documentation

**Review by agutenkunst — APPROVED** [review]

**Pull request merged**